### ISSUE TITLE
ownCloud's interface needs to cater for the needs of advanced users

### ISSUE THREAD

**Issue by oparoz** [owncloud/core#15775]
Copying DropBox seems to be one of the main goals of ownCloud, but that's certainly not the only cloud system out there and while some people prefer dumbed down interfaces, others prefer to use the AWS interface per example, for its flexibility.

One key to a successful user interface is to cater for different [personas](https://en.wikipedia.org/wiki/Persona#In_user_experience_design), but quite often on ownCloud, adding additional interface features are shut down because of the fear of the additional clutter or complexity.

I'm a 100% for an easy to learn interface, but it needs to be able to grow as users improve their knowledge of the system. If one needs to regularly perform specific tasks, he should be able to get them done without having to go through all the baby steps, like a beginner.

You often see in software one of 2 things:
* An advanced or expert mode
* Full interface configuration (Windows, Firefox, etc.)

I'd love to see at least an expert mode in ownCloud

**Comment by biolumin** [comment]
Which baby steps could use enhancement, and how would you suggest we enhance those steps?

**Comment by oparoz** [comment]
I'll give you 2 examples

#### The share dialogue.
It seems every time someone tries to touch it to add functionalities asked for by users, the change is met with resistance. Yes, it has to be kept simple so that it's not overwhelming for beginners, but advanced users don't want to have to switch app, configure special files or worst use a fork to do things like sharing a gallery or document view of their files or to able to enjoy more advanced features.

https://github.com/owncloud/core/issues/8268

#### The slideshow for image previews
Some people have tried to add useful functionalities to the interface and yes, that means having more buttons and some call it clutter, but other users would enjoy those advanced tools. In the end users like @libasys have created a fork to be able to implement their vision.

**Comment by biolumin** [comment]
I understand, and have similar experiences with different open-source communities, so I understand your frustration.

Resistance is to be expected with any reasonable debate, because it's good to see all points of view. Because you referenced topics with a previous thread, I can't comment on those specific issues here. However, I'd suggest you consider coding forks you can add to the ownCloud apps database, or discussing and gaining support from key developers regarding the specific features you cannot code yourself and/or need votes of approval for. It may be a tedious process, but it is the nature of open-source development and community based decision making.

**Comment by oparoz** [comment]
Forks are a lot of work and can be dangerous because it can mean less resources available to both projects, but they're inevitable if a vision shared with a large fraction of the community is incompatible with the one of the owners of the project. We've seen this happening several times with large projects.

But I don't think we're there yet. It's just that I'm not a fan of sacrificing features in the name of simplicity at all cost and would love for oC to think more about different categories of users. Event things which don't seem that advanced are missing from the interface, such as copy/move. It's great that someone has addressed this with an app, but that should come with the default package.

**Comment by biolumin** [comment]
I understand both points of view. I believe we just haven't found the UX/UI option that satisfies both sides of the debate. We'll just have to be patient, and trust the process as well as our ability to solve these complex problems.

FWIW, I'd like additional functionality in this and other areas as well. I'll be putting some time into this and other UX/UI issues soon.

**Comment by oparoz** [comment]
> FWIW, I'd like additional functionality in this and other areas as well. I'll be putting some time into this and other UX/UI issues soon.

I'm glad to hear that :)

**Comment by jospoortvliet** [comment]
For me personally, I don't mind adding more options but only after all other options have been exhausted. And usually, adding another button is done because nobody bothered to find a proper solution to the problem.

If we have a problem, like with the gallery vs file view, let's talk about the problem rather than just discuss a very limited solution to a subset of the problem that introduces its own issues...

Like - we can solve it this way for the gallery app, but what about a folder with music? A folder with documents? We add a new button for each of them?

If the problem is that the files view isn't good at showing some types of files, let's find a solution for that. Either make switching easy (as @oparoz already did adding a button to the files app to switch to the Gallery app in each folder), switch automatically, or make the files app deal better with images. Maybe we don't need a separate Gallery app and we can integrate its view in the Files app.

I'm just throwing up possible solutions but my point is - in ownCloud we've tried to come up with better solutions to problems rather than just 'leave it up to the poor user', as that does not scale. You end up with an UI full of options.

That is different from just saying no to options - although, sometimes, you have to do that too. Some things are useful for less than 0.5% of the users and in that case I say - please, create a fork for those few users who benefit from this, as it simply isn't worth harming the other 99.5% of users for the benefit of few.

For each problem you want to solve, thus, accept that we want to find a proper solution that addresses it rather than working around it by adding another option somewhere; and sometimes, we deem the problem just not important enough to clutter the UI with. If we want a better solution, help by coming up with one. And if it turns out the solution is only going to help very few while making ownCloud more complicated for everybody else (there's more than just UI clutter - maintenance increases with every option too), perhaps it is OK to create a little app that adds the thing you want. Nothing wrong with that (and, mostly, no forks needed).

My way too many cents on the early Saturday morning ;-)
(I mean, dudes, it's before noon!)

**Comment by jancborchardt** [comment]
A lot of it has already been said by @jospoortvliet – thank you.

However I feel I also need to address this since as designer I’m most often the person making these decisions.
Our main goal is definitely _not_ »copying Dropbox« and having a »dumbed down interface« – it is to be useful while being very easy to use. Sometimes that means to make hard decisions, or to just make incremental changes.

In the end we also need to maintain everything we need to introduce. Full interface configuration might be a funny idea – but what for? Everyone who really wants to do this can do this via CSS. And Windows as well as Firefox are not really good examples for configurable interfaces because it’s just a pain. Contrast that to GNOME Shell, OS X or Chrome: No real configuration needed because it’s well designed. And for everything else, there’s 3rd party extensions.
Because experimenting is exactly what the apps ecosystem is for. There people can try stuff out and modify without needing to fork. Because honestly forking ownCloud is not benefitting anyone – it needs to be maintained separately, and will probably not be used by anyone over the main version.

The main thing though is that we need to absolutely focus on quality & stability over features at the moment. I have now worked on this project more than 4 years and there have been very few people complaining about missing features, compared to people complaining about us being more happy to introduce features rather than polish the product.

In general we can not cater to everyone. If we try to design for everyone, then the app will be too bloated for most and unmaintainable. If people like AWS more than Dropbox they just use AWS. If people like something else more than ownCloud they just use that. There are plenty of other open source file storage software projects, but with ownCloud we aim to make stuff simple for people.

**Comment by jancborchardt** [comment]
cc @owncloud/designers if anyone has something to add.

**Comment by biolumin** [comment]
@jancborchardt I resonate with the quality and stability focus. In the environments I use ownCloud in, there are a handful of things that need refinement before resources are poured into new features. 

@oparoz Once these refinements are addresses, I'm sure we'll see some love given to new features we'd all love to see implemented. If not from any one else, like I mentioned, at least from me.

**Comment by jancborchardt** [comment]
@oparoz what do you think about the points from @jospoortvliet and me?

**Comment by libasys** [comment]
My 5 cent to this discussion! The points of @jancborchardt and @jospoortvliet are ok, but for my need i like options and advanced features to enhance owncloud! Meanwhile i have to maintain round about 8 apps, that i can say yes now owncloud is perfect for my needs. So i  am a user/dev of 0.5%! ;) The GUI design is a personal feeling, I am not really happy with the standard design of OC, but this is ok, I can switch to my theme and all is well.

**Comment by oparoz** [comment]
@jospoortvliet 
I agree that we shouldn't implement every idea out there by just adding more and more stuff to the UI, but as you've highlighted, there might be a need to have different views for folders, especially when sharing them, so that they work for the recipient, from the start.
When I suggested we let the sharer pick the view, I was told that this would make the UI too complex. That's where I think there is a problem. Too complex for first time users, probably, but just like some of the other options, it could be revealed when needed and would probably be useful to more than 0.5% of users.

@jancborchardt 
I agree with providing a very easy to use interface, but my point in this discussion is that it doesn't need to be the only one. A home stock broker might use a similar interface to the one a professional broker is using, but the pro will use shortcuts and advanced settings to get the job done quicker and more accurately while using the same backend. Usually these things can't be fixed via CSS only.
It's funny you mention Gnome and OSX. The unilateral thinking, dumbed down interface and hiding as much as possible of the inner working from the user are a few reasons I avoid these types of GUI in favour of KDE and Windows, even if the former are much better than 5-10 years ago. And I don't think the interfaces of the latter are particularly complex, but they give users more control over their experience.
I agree that the focus should be on quality. What's in has to work and additions have to be well integrated, so maybe this is mainly a resources issue, but I don't see ideas and suggestions being tracked for a potential integration at a later date. Maybe that's what hackatons are for and designers go through all the issues together and look for ways to implement them.

To be fair, the Share UI is a good example of an adaptive interface. It's built based on admin settings and is simple to use. I just think it should take into consideration user preferences and offer more options. Same thing with the slideshow example I gave, but that won't happen if one goal of the project is to focus on a single interface for a single group of people.

@libasys
I don't think you're part of such a small minority, but a lot of users probably don't voice their opinion. Some people move on, others will want stability first while people like you will design their own interface.
Ideally you should be able to improve the core interface without having to work on a separate fork, but there has to be a will inside the core team. Another alternative is to release apps which hack the UI and use private methods, but I don't see this as a viable way forward.

(And I think the shipping of a broken official Gallery app in 8.0 is a separate issue)

@biolumin 
I'm looking forward to seeing how the UI will evolve :)

**Comment by jospoortvliet** [comment]
@oparoz I think we can meet in the middle here. KDE follows exactly what I wrote down: try to find a smart solution before you let the user choose. See the changes made since the (loooong ago) release of KDE 3.1 - every release since then has less 'options' but more 'abilities'. Powerful and flexible does not mean full of settings and options, but rather having a smart UX.

I think, when it comes to this specific thing, we need to deal with the problem (the files app is not always the best view on data) rather than only solving this specific thing (sometimes I want to share through the gallery app).

For now, you have a share option in most individual apps which gives you the right view. A related solution might be to make switching to another view easier for the recipient of the data. But let's take time to find a good solution rather than rushing in another button in the UI.

**Comment by oparoz** [comment]
@jospoortvliet - Dolphin still offers incredibly powerful features not found in other popular desktops and they're usually also available via shortcuts, but I do agree that smart is the way to go and that the goal is not to solve every little issue with an option.
Regarding the view in the files app, I see this as a separate issue than wanting to share a link specific to an app. Even if a future version of oC makes it easy to switch view from within the Files app, it should still be possible for a user to create a link which lands in a specific app without having to use the option in the app itself, because the effect of landing on a list of files which may includes documents is not the same as landing on a seamless photo wall.

But that's a very specific case. My main message is that requests for advanced options and interactions should be tracked and worked on (resources permitting).

**Comment by jospoortvliet** [comment]
@oparoz re your main request - advanced options and interactions should sure be tracked and worked on (if people want), but they should be implemented in accordance to our design guidelines/team thoughts and otherwise (if a contributor can't/doesn't want to follow the guidelines or if the feature is simply deemed a bad fit/not needed) can be done as an independent app. That's what I think...

**Comment by oparoz** [comment]
Yes, sensible defaults and hidden advanced settings are key, but as someone has mentioned, this is *own*cloud, not simplecloud. Some users have more advanced needs when it comes to managing/viewing their files and their requests should not automatically be discarded as too complex and irrelevant to most. Apps are defo the way to go, but it's not always possible.

**Comment by jospoortvliet** [comment]
well, where apps can't help deal with this and no sensible solution can be found in the UI, we definitely need to digg deeper. So I agree with you - we shouldn't drop things just because we don't want them in the main user interface. There should be room for everything people want to work on, without having to fork ownCloud - if needed, we have to make changes to our architecture to accommodate that.

But I think we're far from having problems there, so no worries ;-)

**Comment by jancborchardt** [comment]
> this is owncloud, not simplecloud

Meaning you own it does not mean it should be complex. People deserve a cloud of their own which is similarly simple and joyful to use as the proprietary competition.

**Comment by jancborchardt** [comment]
@oparoz I’ll close this issue, ok? It’s a high-level discussion we did, but it’s better to have actionable small issues to work on. :)

**Comment by oparoz** [comment]
Sure :)
